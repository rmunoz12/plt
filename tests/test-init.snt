@setup {

group A(Object) {
    str x;

    func group A __init__(str x) {
        this.x = x;
        return this;
    }

};

}

@turns {

func void begin() {
    int a = 2 + 3;
    int b = a = 5;
    group A g1 = A("hello");
    group A g2;
    group A g3 = g2 = A("world");

    print(a); print("\n");
    print(g1.x + "\n");
    print(g2.x + "\n");
    print(g3.x + "\n");
    g2.x = "!!!";
    print(g2.x + "\n");


    end;
}

}
