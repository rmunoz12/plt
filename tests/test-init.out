5
hello
world
world
!!!
